src/cluster/kv/etcd/store_test.go

```diff
@@ -125,6 +126,28 @@ func TestNoCache(t *testing.T) {
 	require.Equal(t, 0, len(store.(*client).cacheUpdatedCh))
 }
 
+func TestCacheDirCreation(t *testing.T) {
+	ec, opts, closeFn := testStore(t)
+
+	tdir, err := ioutil.TempDir("", "m3tests")
+	defer os.RemoveAll(tdir)
```

Comment:
Should this `defer` be called after `require.NoError`?

Reply:
Fixed